src/go/privileged-service/pkg/port/protproxy.go

```diff
@@ -0,0 +1,161 @@
+/*
+Copyright © 2022 SUSE LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package port
+
+import (
+	"fmt"
+	"net"
+
+	"github.com/rancher-sandbox/rancher-desktop-agent/pkg/types"
+	"k8s.io/kubernetes/pkg/util/netsh"
+	"k8s.io/utils/exec"
+)
+
+type portProxy struct {
+	netsh.Interface
+}
+
+func newPortProxy() *portProxy {
+	return &portProxy{
+		netsh.New(exec.New()),
+	}
+}
+
+func (p *portProxy) execProxy(portMapping types.PortMapping) error {
+	if portMapping.Remove {
+		return p.deleteProxy(portMapping)
+	}
+	return p.addProxy(portMapping)
+
+}
+
+func (p *portProxy) addProxy(portMapping types.PortMapping) error {
+	for k, v := range portMapping.Ports {
+		for _, addr := range v {
+			wslIP, err := getConnectAddr(addr.HostIP, portMapping.ConnectAddrs)
+			if err != nil {
+				return err
+			}
+			args, err := portProxyAddArgs(addr.HostPort, addr.HostIP, k.Port(), wslIP)
+			if err != nil {
+				return err
+			}
+			_, err = p.EnsurePortProxyRule(args)
+			if err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+func (p *portProxy) deleteProxy(portMapping types.PortMapping) error {
+	for _, v := range portMapping.Ports {
+		for _, addr := range v {
+			args, err := portProxyDeleteArgs(addr.HostPort, addr.HostIP)
+			if err != nil {
+				return err
+			}
+			err = p.DeletePortProxyRule(args)
+			if err != nil {
```

Comment:
After discussion with Mark, no changes here are needed since the error here is not terminating the process but rather we are emitting a service exception log in the windows event log.